src/Kiota.Builder/Refiners/GoRefiner.cs

```diff
@@ -421,6 +424,33 @@ private void FlattenGoFileNames(CodeElement currentElement)
         CrawlTree(currentElement, FlattenGoFileNames);
     }
 
+    private static void FixConstructorClashes(CodeElement currentElement, Func<string, string> nameCorrection)
+    {
+        var allClasses = new List<CodeClass>();
+        FindAllClasses(currentElement, allClasses);
+
+        foreach (var codeClass in allClasses)
+        {
+            if (codeClass.Name.StartsWith("New", StringComparison.OrdinalIgnoreCase))
+            {
+                var targetName = codeClass.Name.Substring(3);
+                if (allClasses.Exists(x => x.Name.Equals(targetName, StringComparison.OrdinalIgnoreCase)))
+                    codeClass.Name = nameCorrection(codeClass.Name);
```

Comment:
I'd rather use the rename method from the parent to avoid introducing potential fork trees.

Reply:
👍